mlflow/h2o.py

```diff
@@ -24,49 +39,62 @@ def save_model(h2o_model, path, conda_env=None, mlflow_model=Model(), settings=N
 
     :param h2o_model: H2O model to be saved.
     :param path: Local path where the model is to be saved.
+    :param conda_env: Path to a Conda environment file. If provided, this decribes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in ``mlflow.h2o.DEFAULT_CONDA_ENV``. If `None`, the default
+                      ``mlflow.h2o.DEFAULT_CONDA_ENV`` environment will be added to the model.
```

Comment:
Is there any way we could make this argument accept either a file or a Python object eventually? As @smurching found with the Databricks cluster spec object, sometimes it's annoying to write a file and people would prefer to pass in a string with the content. We could add it later but I was thinking we might want to just add it up front somehow.

Reply:
Some options are:
- You are allowed to pass in a dictionary object that we then turn to YAML. If the argument is a dict, we do that, otherwise we treat it as as string and read that file.
- You always have to pass in a dictionary (if you had a conda.yaml, read it into a dictionary first).

Reply:
@mateiz yeah it's a good point, I think the main change needed would be to change the logic [here](https://github.com/mlflow/mlflow/pull/708/files#diff-3a00bc8af91e17c6ec67ad5944cc5f1fR69) to handle values that aren't paths to files. We'd need to make that change across each model type + also add a test, so my inclination would be to get this PR in and add support for passing dicts to `conda_env` in a follow-up PR.